### PR TITLE
Inclusion of energy_reco and energy_true in MapMaker

### PR DESCRIPTION
**Pull request by AtreyeeS**
This supersedes PR #1579.

### CODE REVIEW

**Comment by cdeil** [comment]
@AtreyeeS - This PR changes something about the energy axis, and it copies the class to a new file. This makes the diff very hard to review. Do you still have time to send a PR with only the energy axis change?

File: gammapy/cube/mapmaker.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    ----------
+
+    ref_geom_ereco : `~gammapy.maps.WcsGeom`
+        Reference image geometry for the counts and background maps in reco energy
+    ref_geom_etrue : `~gammapy.maps.WcsGeom`
+
+    It is assumed that ref_geom_ereco and ref_geom_etrue have the same wcs but can have different
+    energy binning.
+
+    TODO: 1. Can they have different WCS and we do a reproject?
```
Comment (cdeil) [inline gammapy/cube/mapmaker.py]: @AtreyeeS - This `MapMaker` docstring is no longer properly formatted. If you run `python setup.py build_docs` locally and look at the HTML docs you will see that. If you want to keep these TODOs are reminder issues, please put them as code comments.

You can assume that all the WCS are the same, users should just reproject before using the MapMaker if needed.
We also shouldnt need two geom to specify this, maybe just take an extra energy Quantity vector, default to None and in that case make the two the same?

File: gammapy/cube/tests/test_new.py [hunk 2]
```diff
@@ -107,9 +108,10 @@ def test_MapMaker(mode, expected):
     assert mmaker.exposure_map.unit == "m2 s"
     assert_quantity_allclose(mmaker.counts_map.data.sum(), expected)
 
-    maker = MapMaker(geom, 6.0 * u.deg, cutout_mode=mode)
+    etrue_axis = MapAxis.from_bounds(0.5, 50.0, 10, name='energy', unit='TeV', interp='log')
+    geom_etrue = WcsGeom.create(binsz=0.1 * u.deg, skydir=pos_SagA, width=15.0, axes=[etrue_axis])
+    maker = MapMaker(geom, offset_max=6.0 * u.deg, ref_geom_etrue=geom_etrue, cutout_mode=mode)
```
Comment (cdeil) [inline gammapy/cube/tests/test_new.py]: If there's a change in the code and test, there should also be a change in an assert.
Can you please add a useful assert that establishes something about this change in energy axis?

**Comment by AtreyeeS** [comment]
Hello... sorry I really won't get any time before Thursday! Closing this PR, and thanks for taking this up!

**Pull request closed without merging**